code/datums/datacore.dm

```diff
@@ -251,7 +256,166 @@ var/record_id_num = 1001
 		L.fields["b_dna"]		= H.dna.unique_enzymes
 		L.fields["enzymes"]		= H.dna.struc_enzymes
 		L.fields["identity"]	= H.dna.uni_identity
-		L.fields["image"]		= getFlatIcon(H)	//This is god-awful
+		L.fields["image"]		= image
 		locked += L
 	return
 
+/obj/effect/datacore/proc/get_id_photo(var/mob/living/carbon/human/H)
+	var/icon/photo = null
+	var/g = (H.gender == FEMALE) ? "f" : "m"
+	if(!config.mutant_races || H.dna.species.use_skintones)
+		photo = icon("icon" = 'icons/mob/human.dmi', "icon_state" = "[H.skin_tone]_[g]_s")
+	else
+		photo = icon("icon" = 'icons/mob/human.dmi', "icon_state" = "[H.dna.species.id]_[g]_s")
+		photo.Blend("#[H.dna.mutant_color]", ICON_MULTIPLY)
+
+	var/icon/eyes_s
+	if(EYECOLOR in H.dna.species.specflags)
+		eyes_s = icon("icon" = 'icons/mob/human_face.dmi', "icon_state" = "[H.dna.species.eyes]_s")
+		eyes_s.Blend("#[H.eye_color]", ICON_MULTIPLY)
+
+	var/datum/sprite_accessory/S
+	S = hair_styles_list[H.hair_style]
+	if(S && (HAIR in H.dna.species.specflags))
+		var/icon/hair_s = icon("icon" = S.icon, "icon_state" = "[S.icon_state]_s")
+		hair_s.Blend("#[H.hair_color]", ICON_MULTIPLY)
+		eyes_s.Blend(hair_s, ICON_OVERLAY)
+
+	S = facial_hair_styles_list[H.facial_hair_style]
+	if(S && (FACEHAIR in H.dna.species.specflags))
+		var/icon/facial_s = icon("icon" = S.icon, "icon_state" = "[S.icon_state]_s")
+		facial_s.Blend("#[H.facial_hair_color]", ICON_MULTIPLY)
+		eyes_s.Blend(facial_s, ICON_OVERLAY)
+
+	if(eyes_s)
+		photo.Blend(eyes_s, ICON_OVERLAY)
+
+	var/icon/clothes_s = null
+	switch(H.mind.assigned_role)

```

Comment:
This is so long I would actually recommend optimizing the order this gets evaluated in so more likely jobs are checked sooner (more likely to have multiple engineers than Research Directors or HoPs).